lib/ansible/modules/monitoring/grafana_datasource.py

```diff
@@ -0,0 +1,469 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017, Thierry Sallé (@tsalle)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.1'
+}
+
+DOCUMENTATION = '''
+---
+module: grafana_datasource
+author:
+  - "Thierry Sallé (@tsalle)"
+version_added: "2.5"
+short_description: Manage grafana datasources
+description:
+  - Create/update/delete grafana datasources via API
+options:
+  grafana_url:
+    required: true
+    description:
+      - Grafana url
+  name:
+    required: true
+    description:
+      - Name of the datasource
+  ds_type:
+    required: true
+    choices: [graphite,prometheus,elasticsearch,influxdb,opentsdb]
+    description:
+     - Type of the datasource C(graphite),C(prometheus),C(elasticsearch),C(influxdb),C(opentsdb)
+  url:
+    required: true
+    description:
+      - Url of the datasource
+  access:
+    required: false
+    default: proxy
+    choices: [proxy,direct]
+    description:
+      - Access mode for this datasource
+  grafana_user:
+    required: false
+    default: admin
+    description:
+      - Grafana API user
+  grafana_password:
+    required: false
+    default: admin
+    description:
+      - Grafana API password
+  grafana_api_key:
+    required: false
+    description:
+      - Grafana API key
+      - If set, grafana_user and grafana_password will be ignored
+  database:
+    required: false
+    description:
+      - Name of the database for the datasource.
+      - This options is required when the ds_type is influxdb or elasticsearch (index name)
```

Comment:
Should be enforced by argspec

Reply:
the `tls_ca_cert` is not mandatory when tls_client auth is enabled.
It's just an other option to trust the certificate of the datasource.

Reply:
done with `required_if`